lib/SpeciesTreeBuilder/SpeciesTreeBuilderClient.pm

```diff
@@ -81,20 +81,19 @@ sub new
     # We create an auth token, passing through the arguments that we were (hopefully) given.
 
     {
-	my $token = Bio::KBase::AuthToken->new(@args);
+	my %arg_hash2 = @args;
+	if (exists $arg_hash2{"token"}) {
+	    $self->{token} = $arg_hash2{"token"};
+	} elsif (exists $arg_hash2{"user_id"}) {
+	    my $token = Bio::KBase::AuthToken->new(@args);
+	    if (!$token->error_message) {
+	        $self->{token} = $token->token;
+	    }
+	}
 	
-	if (!$token->error_message)
+	if (exists $self->{token})
 	{
-	    $self->{token} = $token->token;
-	    $self->{client}->{token} = $token->token;
-	}
-        else
-        {
-	    #
-	    # All methods in this module require authentication. In this case, if we
-	    # don't have a token, we can't continue.
-	    #
-	    die "Authentication failed: " . $token->error_message;
+	    $self->{client}->{token} = $self->{token};
```

Comment:
You should handle the case of no token.

Reply:
That client code is auto-generated by the SDK.  I'm using SDK version 1.0.15 (commit 3471b17d8b69fa54ed44b6784262d1a2948c2283).  Is there something better generated by newer SDK versions that I should use instead?